### PR TITLE
AWS: Add Shared VPC workflow & periodic

### PR DESCRIPTION
**Pull request by patrickdillon**

### CODE REVIEW

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml [hunk 1]
```diff
@@ -970,6 +970,14 @@ tests:
   steps:
     cluster_profile: gcp-openshift-gce-devel-ci-2
     workflow: openshift-e2e-gcp-graceful-shutdown
+- as: e2e-aws-ovn-shared-vpc-phz-techpreview
+  interval: 8h
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
```
Comment (patrickdillon) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:978]: I'm going to check out how we can move this to use the aws cluster-profile.

**Comment by patrickdillon** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-shared-vpc-phz-techpreview

**Comment by patrickdillon** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-shared-vpc-phz-techpreview

**Comment by patrickdillon** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-shared-vpc-phz-techpreview

File: ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh [hunk 2]
```diff
@@ -68,12 +68,13 @@ if [[ ${ENABLE_SHARED_PHZ} == "yes" ]]; then
...
 
   role_name=$(head -n 1 ${SHARED_DIR}/shared_install_role_name)
-  policy_arn=$(head -n 1 ${SHARED_DIR}/shared_install_policy_arn)
+  shared_policy_arn=$(head -n 1 ${SHARED_DIR}/shared_install_policy_arn)
 
   aws_delete_role $REGION ${role_name}
   echo "Deleted role: ${role_name}"
 
-  aws --region $REGION iam delete-policy --policy-arn ${policy_arn}
+  echo "Deleting policy: ${policy_arn}"
```
Comment (yunjiang29) [inline ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh:76]:
```suggestion
  echo "Deleting policy: ${shared_policy_arn}"
```

Comment (yunjiang29) [inline ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh:76]:
```suggestion
  echo "Deleting policy: ${shared_policy_arn}"
```

File: ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh [hunk 3]
```diff
@@ -68,12 +68,13 @@ if [[ ${ENABLE_SHARED_PHZ} == "yes" ]]; then
...
+  aws --region "$REGION" iam delete-policy --policy-arn "${shared_policy_arn}"
   echo "Deleted policy: ${policy_arn}"
```
Comment (yunjiang29) [inline ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh:78]:
```suggestion
  echo "Deleted policy: ${shared_policy_arn}"
```

**Comment by jianlinliu** [comment]
@patrickdillon I suggest to add one more line in `aws_delete_role` function to avoid duplicate variables name overriding issue for more guards.
```
local policy_arn attached_policies inline_policies policy_name
```

**Comment by deepsm007** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-shared-vpc-phz-techpreview

**Pull request merged**